AdyenEncryption/JOSE/Encryption Algorithms/Content Encryption Algorithms/JWAA256CBCHS512Algorithm.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import CommonCrypto
+import Foundation
+
+internal struct JWAA256CBCHS512Algorithm: JWAEncryptionAlgorithm {
+    internal let keyLength: Int = kCCKeySizeAES256 * 2
+    
+    internal let initializationVectorLength: Int = kCCBlockSizeAES128
+    
+    internal let name: String = "A256CBC-HS512"
+    
+    internal func encrypt(input: JWAInput) throws -> JWAOutput {
+        guard input.key.count == keyLength else {
+            throw EncryptionError.invalidKey
+        }
+        guard input.initializationVector.count == initializationVectorLength else {
+            throw EncryptionError.invalidInitializationVector
+        }
+
+        let (aesKey, hmacKey) = deriveKeys(from: input.key)
+        
+        let operation = CCOperation(kCCEncrypt)
+        let option = CCOptions(kCCOptionPKCS7Padding)
```

Comment:
We haven't use PKCS7Padding previously 🤔

Reply:
as discussed separately, padding schemes are just so that the cypher text is an integer multiple of the block size. it doesn't alter the behavior of the encryption or authentication schemes. and PKCS7 is a pretty solid padding scheme.

Reply:
it is a recommended padding scheme for AES